vault/plugin_catalog.go

```diff
@@ -61,7 +62,7 @@ type PluginClient struct {
 	// client handles the lifecycle of a plugin process
 	// multiplexed plugins share the same client
 	client      *plugin.Client
-	protocol    plugin.ClientProtocol
+	protocol    *pluginClientConn
```

Comment:
Maybe instead this could have both `protocol` and `conn`? Like so

```go
protocol    plugin.ClientProtocol
conn        *pluginClientConn
```

I opted to just store `protocol` since it can derive `conn`.

Reply:
Yeah, I think just having `*pluginClientConn` is fine.

Reply:
Would it make sense to name the field, `clientConn` instead?

Reply:
We do still need ClientProtocol which is why I went with deriving the conn as my first approach. But I think it makes more sense to just store both of them.

Reply:
addressed in: https://github.com/hashicorp/vault/pull/13896/commits/7c1e08b56e41a2259c1550ea83d767f8a52bb6b5